slang-api/src/main/java/com/sonarsource/slang/impl/TreeMetaDataProvider.java

```diff
@@ -61,5 +64,13 @@ public TextRange textRange() {
         .filter(comment -> comment.textRange().isInside(textRange))
         .collect(Collectors.toList());
     }
+
+    @Override
+    public List<Token> tokens() {
+      // TODO improve performance by storing an ordered list of tokens
```

Comment:
Another possible improvement would be to have the tokens/comments computed only once and stored in the `TreeMetaDataImpl` so that it doesn't recompute it when called multiple times